### PR TITLE
Add supported types with charset

### PR DESCRIPTION
**Pull request by y-tang**
@shelbys @joelteon @tfemiani review please

### CODE REVIEW

File: lib/http-context.js [hunk 1]
```diff
@@ -310,10 +314,15 @@ HttpContext.prototype.done = function () {
   if (this.req.query._format) {
     accepts = this.req.query._format.toLowerCase();
   }
-  var dataExists = typeof data !== 'undefined';
 
+  var acceptableMIMEType = accepts;
+  if (acceptableMIMEType) {
+    acceptableMIMEType = acceptableMIMEType.replace(/;\s*charset=[^;]+;*/i, '');
```
Comment (shelbys) [inline lib/http-context.js]: Shouldn't this be `acceptableMIMEType.split(';')[0]` to ensure we just throw away whatever specializations are present?

**Comment by shelbys** [comment]
@y-tang - Done reviewing

File: lib/http-context.js (hunk 1 above)
Comment (tfemiani) [inline lib/http-context.js]: Maybe something in between, the `Accept` header can have multiple content types specified, comma separated. Each one can have specializations. Does the code that we are modifying handle multiple content types in the `Accept` header already?

Comment (y-tang) [inline lib/http-context.js]: @shelbys There might be other legitimate parameters, like `level=1`. Do we want to throw that away too?

Comment (y-tang) [inline lib/http-context.js]: @tfemiani The returned `accepts` from `this.req.accepts(this.supportedTypes)` will have only one winning media type

**Comment by shelbys** [comment]
:+1:

**Pull request merged**